SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizer.cc

```diff
@@ -93,17 +93,14 @@ namespace cms
     mixMod.produces<edm::DetSetVector<Phase2TrackerDigi> >("Tracker").setBranchAlias(alias2);
     mixMod.produces<edm::DetSetVector<PixelDigiSimLink> >("Tracker").setBranchAlias(alias2);
 
+    // creating algorithm objects and pushing them into the map
+    algomap_[AlgorithmType::InnerPixel] = std::unique_ptr<Phase2TrackerDigitizerAlgorithm>(new PixelDigitizerAlgorithm(iconfig_));
+    algomap_[AlgorithmType::PixelinPS]  = std::unique_ptr<Phase2TrackerDigitizerAlgorithm>(new PSPDigitizerAlgorithm(iconfig_));
+    algomap_[AlgorithmType::StripinPS]  = std::unique_ptr<Phase2TrackerDigitizerAlgorithm>(new PSSDigitizerAlgorithm(iconfig_));
+    algomap_[AlgorithmType::TwoStrip]   = std::unique_ptr<Phase2TrackerDigitizerAlgorithm>(new SSDigitizerAlgorithm(iconfig_));
```

Comment:
Right-hand side could be written more shortly as `std::make_unique<PixelDigitizerAlgorithm>(iConfig);` etc.

On a more general note (possibly going beyond this PR)
* `algomap_` could be implemented as `std::vector` (consumes less memory, faster access)
* `iconfig_` could now be removed if `isOTreadoutAnalog` parameter would be read to member variable